### ISSUE TITLE
Empty SMTP FROM for successful delivery notification message

### ISSUE THREAD

**Issue by paulmenzel** [sympa-community/sympa#1843]
Using Debian 12 (stable/bookworm) with *sympa* 6.2.70~dfsg-2 and *exim4-daemon-heavy* 4.96-15+deb12u4, a non-subscriber with a Google Mail (gmail) address sent a message, that got moderated. After accepting the message, a delivery confirmation message was to be sent, but got stuck in the queue, because Exim tries to send it with an empty envelope from:

Incoming message:

    Received: from mail-yw1-x1146.google.com ([2607:f8b0:4864:20::1146]:47471)
        by mail.gw90.de with esmtps  (TLS1.3) tls TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
        (Exim 4.96)
        (envelope-from <3YRZPZgoJBgQvmwglywxD4kqemp.gsqmrxivrzsgerxevi.hi@trix.bounces.google.com>)
        id 1sA5QI-00GjwC-2P
        for mylist@vocantare.de;
        Thu, 23 May 2024 9:00:00 +0000

The notification message about the moderation (*Nachrichtenverteilung*) is then successfully sent:

    2024-05-23 9:00:10.652 [3989855] 1sA5QM-00GjwQ-2c => ab@gmail.com F=<mylist-owner@vocantare.de> P=<mylist-owner@vocantare.de> R=dnslookup T=remote_smtp S=658 H=gmail-smtp-in.l.google.com [2a00:1450:4013:c1a::1b]:25 I=[2a01:4f8:200:641c::d0d0:c0de]:54588 TFO X=TLS1.3:ECDHE_X25519__ECDSA_SECP256R1_SHA256__AES_256_GCM:256 CV=yes DN="CN=mx.google.com" L K C="250 2.0.0 OK <somehash> - gsmtp" QT=0.836s DT=0.675s

But the message about the successful delivery does not have the F= field set, and therefore Google Mail rejects it:

    2024-05-23 10:00:00.000 [3991450] 1sA5sL-00GkMC-2O ** ab@gmail.com F=<> P=<> R=dnslookup T=remote_smtp H=gmail-smtp-in.l.google.com [2a00:1450:4013:c1a::1a]:25 I=[2a01:4f8:200:641c::d0d0:c0de]:59848 X=TLS1.3:ECDHE_X25519__ECDSA_SECP256R1_SHA256__AES_256_GCM:256 CV=yes DN="CN=mx.google.com": SMTP error from remote mail server after end of data: 550-5.7.26 Your email has been blocked because the sender is unauthenticated.\n550-5.7.26 Gmail requires all senders to authenticate with either SPF or DKIM.\n550-5.7.26\n550-5.7.26  Authentication results:\n550-5.7.26  DKIM = did not pass\n550-5.7.26  SPF [] with ip: [2a01:4f8:200:641c::d0d0:c0de] = did not pass\n550-5.7.26\n550-5.7.26  For instructions on setting up authentication, go to\n550 5.7.26  https://support.google.com/mail/answer/81126#authentication a640c23a62f3a-a5a17b20fdcsi1619192466b.431 - gsmtp DT=0.458s

**Comment by ikedas** [comment]
The rejection message by recipient (google.com) is:
```
550-5.7.26 Your email has been blocked because the sender is unauthenticated.
550-5.7.26 Gmail requires all senders to authenticate with either SPF or DKIM.
550-5.7.26
550-5.7.26  Authentication results:
550-5.7.26  DKIM = did not pass
550-5.7.26  SPF [] with ip: [2a01:4f8:200:641c::d0d0:c0de] = did not pass
550-5.7.26
550-5.7.26  For instructions on setting up authentication, go to
550 5.7.26  https://support.google.com/mail/answer/81126#authentication a640c23a62f3a-a5a17b20fdcsi1619192466b.431 - gsmtp DT=0.458s
```
Follow the link above and read the description by Google.  They require authentication at least by either SPF or DKIM, or both.  However your site provides neither.

Your options are either:
*   Setting Sympa to add DKIM signature.
    See the [documentation](https://www.sympa.community/manual/customize/dkim-arc.html).

*   Setting DNS resource record for SPF.
    A common mistake is not setting a SPF record on the smtp.helo host for the DSNs that has empty envelope sender.  Check the [description](http://www.open-spf.org/action_browse_id_FAQ/Common_mistakes_revision_26/#helo).  In fact, there is no SPF record for your SMTP HELO host mail.gw90.de.

**Comment by paulmenzel** [comment]
@ikedas, thank you for looking into this. That was my first thought too, but then I wondered, why do all the other messages pass. If envelope-from is empty, then whose SPF record should be checked, and that is the problem in my opinion. This is substantiated by looking at [other rejection messages](https://support.google.com/a/thread/239644302/gmail-requires-all-senders-to-authenticate-with-either-spf-or-dkim?hl=en):

    SPF [jjworldtrade.com] with ip: …

So Google Mail tells the domain, for which the SPF record is missing. In my case it’s the empty string.

**Comment by paulmenzel** [comment]
And for the record, SPF records are set up:

    $ dig txt vocantare.de +short | grep spf
    "v=spf1 mx ~all"
    $ dig mx vocantare.de +short
    10 mail.gw90.de.

**Comment by ikedas** [comment]
No, please read my explanation and description on open-spf.org carefully.

SPF record should be set up on SMTP HELO/EHLO host. Expected result is:
```
$ dig txt mail.gw90.de +short
"v=spf1 (...appropriate SPF specifications...) ~all"
```

**Comment by paulmenzel** [comment]
Please explain the Google Mail error message (empty []) and the Exim logs, why envelope sender is not set.

**Comment by ikedas** [comment]
Why the envelope sender of DSN is empty is that it must be empty.  [RFC 3464](https://datatracker.ietf.org/doc/html/rfc3464) says: _Whenever an SMTP transaction is used to send a DSN, the MAIL FROM command MUST use a NULL return address, i.e., "MAIL FROM:<>"_. Sympa merely conforms to this.  Moreover, your Exim also sets envelope sender to be null when it originates DSN: Please check the logs.

[RFC 7208](https://datatracker.ietf.org/doc/html/rfc7208) says: _In this [Delivery Status Notification] case, the only entity available for performing an SPF check is the "HELO" identity_.

That't why you should set up SPF record for SMTP HELO/EHLO host name, not only the email domain name.

**Comment by paulmenzel** [comment]
Thank you for going into the details. I set this up now, so let’s see if it works next time.

    $ host -t txt mail.gw90.de
    mail.gw90.de descriptive text "v=spf1 a -all"

Thank you, thank you, again.

**Comment by ldidry** [comment]
If anyone comes here after encountering the same problem that make me look at this closed issue, Microsoft recently began to reject our DSN mails with the following message:

```
550 5.7.509 Access denied, sending domain [OUR_DOMAIN.ORG] does not pass DMARC verification and has a DMARC policy of reject. [VI1P195MB0734.EURP195.PROD.OUTLOOK.COM 2025-06-23T20:38:28.040Z 08DDB2564DE36667] [DU2P250CA0022.EURP250.PROD.OUTLOOK.COM 2025-06-23T20:38:28.060Z 08DDB2063B7A7D58] [DU6PEPF00009523.eurprd02.prod.outlook.com 2025-06-23T20:38:28.063Z 08DDB1257591DF93] (in reply to end of DATA command))
```

We have a SPF, DKIM and DMARC configuration / DNS records, all is good, our mails are accepted by Microsoft except for our "rejected because too large" DSN mails (and maybe other DSN mails, but that’s all the DSN mails I found when debugging the problem).

TL;DR: it seems that Microsoft does not know how to correctly check mails with empty FROM.
